programs/nft-voter/src/instructions/configure_collection.rs

```diff
@@ -0,0 +1,104 @@
+use anchor_lang::{
+    account,
+    prelude::{Context, Signer},
+    Accounts,
+};
+
+use anchor_lang::prelude::*;
+use anchor_spl::token::Mint;
+use spl_governance::state::realm;
+
+use crate::state::{CollectionConfig, Registrar};
+use crate::{error::NftVoterError, state::MaxVoterWeightRecord};
+
+/// Configures NFT voting collection which defines what NFTs can be used for governances
+/// and what weight they have
+/// The instruction updates MaxVoterWeightRecord which is used by spl-gov to determine max voting power
+/// used to calculate voting quorum    
+#[derive(Accounts)]
+pub struct ConfigureCollection<'info> {
+    /// Registrar for which we configure this Collection
+    #[account(mut,
+        constraint = registrar.realm == realm.key() 
+        @ NftVoterError::InvalidRegistrarRealm
+    )]
+    pub registrar: Account<'info, Registrar>,
+
+    /// CHECK: Owned by spl-governance instance specified in registrar.governance_program_id
```

Comment:
So you can actually check this using
`#[account(owner = registrar.governance_program_id)]`
I recommend doing this for all the accounts you expect to be owned by the governance program

Reply:
The owner checks are redundant here because the deserialisation functions from `spl-gov` always check the owner. However I agree it's more idiomatic to `Anchor` syntax and I added them.

Yeah, I want to go super fancy with it and I'll give it a try. It should work as long as the desrialisation can ignore `owner` which is only known at runtime, but I think Anchor always requires that so it's not possible.
The use case of a multi instance programs is not supported by `Anchor` at all atm.

I also run into an issue with an optional account. The owner check panics if the account is missing. Is there a solution for that?